agent/job_runner.go

```diff
@@ -488,6 +494,10 @@ func (r *JobRunner) createEnvironment(ctx context.Context) ([]string, error) {
 	env["BUILDKITE_STRICT_SINGLE_HOOKS"] = fmt.Sprintf("%t", r.conf.AgentConfiguration.StrictSingleHooks)
 	env["BUILDKITE_SIGNAL_GRACE_PERIOD_SECONDS"] = fmt.Sprintf("%d", int(r.conf.AgentConfiguration.SignalGracePeriod/time.Second))
 
+	if r.conf.KubernetesExec {
+		env["BUILDKITE_KUBERNETES_EXEC"] = "true"
```

Comment:
This seems to be new. Just for my curiosity, what is this impacting?

Reply:
Previously, if `kubernetes-exec` were enabled (as an experiment), it would be passed down to the subprocess via the `BUILDKITE_AGENT_EXPERIMENTS` variable. Now it could be set in the parent process via a flag, we should ensure that if enabled it should be passed to the subprocess. 

But we'll make the k8s controller set this anyway, so it's probably unnecessary.